tests/jobs/scala/src/main/scala/sorting/SortingApp.scala

```diff
@@ -1,18 +1,48 @@
 package sorting
 
+import java.io.{BufferedReader, InputStreamReader}
+
+import org.apache.spark.input.PortableDataStream
 import org.apache.spark.sql.SparkSession
 
+import scala.collection.SortedSet
+import scala.collection.JavaConverters._
+import scala.collection.mutable.ListBuffer
+import scala.reflect.ClassTag
+import scala.util.matching.UnanchoredRegex
+
 /**
-	* Performs sort (transformation) and count (action) on the Dataset, loaded from [s3bucket].
+	* Performs sort (transformation) and count (action) on the Dataset, loaded from [input path].
+	* In case [input path] requires authentication, all the properties should be provided via
+	* Spark Configuration or spark-submit '--conf key=value' arguments.
 	*
-	* Usage: SortingApp [s3bucket]
+	* Usage: SortingApp [input path] [output path]
 	*/
 
 object SortingApp {
 
+	val pattern: UnanchoredRegex = """"key":"([a-zA-Z0-9]+)"""".r.unanchored
+
+	case class PartitionValidationResult(
+																				fileName: String,
+																				isSuccess: Boolean,
+																				totalKeys: Long,
+																				outOfOrderKeys: Long,
+																				sample: List[String],
+																				firstKey: String,
+																				lastKey: String) extends Ordered[PartitionValidationResult] {
+
+
+		override def toString: String = {
+			s"PartitionValidationResult {\n\tfile: $fileName,\n\tsuccess: $isSuccess,\n\ttotalKeys: $totalKeys,\n\tfirst: $firstKey,\n\tlast: $lastKey,\n\tunorderedKeys: $outOfOrderKeys,\n\tsample:\n${sample.mkString("\n")}}"
```

Comment:
Maybe multi-line string here for readability? Eg
```
"""
PartitionValidationResult {
  file: $fileName,
  ...
"""
```

Reply:
good point, updated